nearform/github-action-notify-release
# Feat/393 post comment when issue is closed as not planned

It closes #393 

### Main changes:

- added a new trigger on `issues` `[closed]` in the Github action workflow
- added a new check in the main file so that we can handle the new trigger only if these conditions are met:
    - `event_name == 'issues' && state == 'closed' && state_reason == 'not_planned' && issue_label == 'notify-release'`
- if the above conditions are met the new `addComment` functionality will be executed.
- added unit tests to reach 100% code coverage.

### Tests:

Below you can find the list of tests I performed to check the new feature:

1. run manually the action: a new issue is created as expected :white_check_mark:
2. run the action after a release has been published: a new issue is created only if new commits are present :white_check_mark:
3. close the issue as complete: the action is triggered but nothing happens, because we only handle the closing event on `not_planned` :white_check_mark:
4. close the issue as not_planned: the action is triggered and a new comment is added to the issue :white_check_mark:

<details>
<summary>Test screenshots:</summary>

1. run manually the action
<img width="425" alt="image" src="https://user-images.githubusercontent.com/6604111/215691063-45eb5b53-bd8f-40a8-9c73-c45724825049.png">

2. run the action after a release has been published
<img width="1050" alt="image" src="https://user-images.githubusercontent.com/6604111/215691514-d5cd66e6-9ab7-43e1-8588-2ddf816a6b7d.png">

3. close the issue as complete
<img width="1041" alt="image" src="https://user-images.githubusercontent.com/6604111/215690276-ac29c939-4a42-48ff-a6e0-094fd09b1570.png">

4. close the issue as not_planned
<img width="1045" alt="image" src="https://user-images.githubusercontent.com/6604111/215690851-e15ac20d-6edf-4db2-a0ef-0ccf37489ea8.png">

<img width="916" alt="image" src="https://user-images.githubusercontent.com/6604111/215690940-18d83d78-3ea0-4dd5-abb2-6de335e64ff1.png">
</details>
